larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.cc

```diff
@@ -36,19 +55,111 @@ void TwoViewTransverseTracksAlgorithm::CalculateOverlapResult(const Cluster *con
     if ((xSpan1 < std::numeric_limits<float>::epsilon()) || (xSpan2 < std::numeric_limits<float>::epsilon()))
         throw StatusCodeException(STATUS_CODE_INVALID_PARAMETER);
 
-    // TODO Create a fully-fledged overlap result object and fill it with useful information
-    const float xOverlap(std::min(xMax1, xMax2) - std::max(xMin1, xMin2));
+    const float xOverlapMin(std::max(xMin1, xMin2));
```

Comment:
I don't think the `TwoViewXOverlap` object performs any comparisons, it copies the arguments verbatim.
However, it --could-- do the comparisons during construction.

The three-view case operates in an identical manner (I largely copied the structure).

I am happy to change the `TwoViewXOverlap` constructor to do the calculations to figure out the min/max and span, but we might also want to consider changing the three view case as well.

Reply:
Alright, I've given it a go.
The TwoViewXOverlap class now calculates the overlap in its constructor initialiser list.  It does not (and actually cannot) do any kind of sanity checking, such as checking if the individual view spans are > 0, because there are default constructions of the class where all of the min/max are set to 0.  The sanity checks are still done in the TransverseMatchingAlg.
I'll leave this comment open for a bit.